batch7crud-roll6/src/main/java/com/lftechnology/batch7crud/dao/UserDAOImpl.java

```diff
@@ -0,0 +1,143 @@
+            String str = "insert into user (firstname,surname,username,password) values (?,?,?,?)";
+            PreparedStatement statement = connection.prepareStatement(str);
+
+            statement.setString(1, user.getFirstName());
+            statement.setString(2, user.getSurName());
+            statement.setString(3, user.getUserName());
+            statement.setString(4, user.getPassword());
+
+            statement.execute();
+            System.out.println("user added");
+        } catch (SQLException e) {
+            System.err.println(e);
+            throw new DataException();
+        }
+    }
+
+    public void delete(int userID) throws DataException {
+        try {
+            String query = "delete from user where id= ?";
+            PreparedStatement statement = connection.prepareStatement(query);
+
+            statement.setLong(1, userID);
+            statement.executeUpdate();
+
+            System.out.println("user deleted");
+        } catch (SQLException e) {
+            System.err.println(e);
+            throw new DataException(e.getMessage());
+        }
+    }
+
+    public List<User> fetch(int page) throws DataException {
+        String query = "select * from user";
+
+        List<User> userList = new ArrayList<User>();
+        PreparedStatement statement;
+        try {
+            statement = connection.prepareStatement(query);
+            ResultSet results = statement.executeQuery();
+
+            User user = null;
+
+            while (results.next()) {
+                user = new User();
+
+                user.setId(TypeCaster.toInt(results.getString("id")));
+                user.setFirstName(results.getString("firstname"));
+                user.setSurName(results.getString("surname"));
+                user.setUserName(results.getString("username"));
+                user.setPassword(results.getString("password"));
+
+                userList.add(user);
+
+            }
+
+        } catch (SQLException e) {
+            System.err.println(e);
+            throw new DataException(e.getMessage());
+        }
+        return userList;
+
+    }
+
+    public User fetchByID(int userID) throws DataException {
+        User user = null;
+
+        try {
+            String query = "select * from user where id=?";
+            PreparedStatement statement = connection.prepareStatement(query);
+
+            statement.setLong(1, userID);
+            ResultSet results = statement.executeQuery();
+
+            while (results.next()) {
+                user = new User();
+
+                user.setId(TypeCaster.toInt(results.getString("id")));
+                user.setFirstName(results.getString("firstname"));
+                user.setSurName(results.getString("surname"));
+                user.setUserName(results.getString("username"));
+                user.setPassword(results.getString("password"));
+            }
+
+        } catch (SQLException e) {
+            System.err.println(e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)